src/Psalm/Type/Atomic.php

```diff
@@ -116,7 +151,32 @@ public static function create(
         string $value,
         ?int   $analysis_php_version_id = null,
         array  $template_type_map = [],
-        array  $type_aliases = []
+        array  $type_aliases = [],
+        ?int   $offset_start = null,
+        ?int   $offset_end = null,
+        ?string $text = null,
+        bool    $from_docblock = false
+    ): Atomic {
+        $result = self::createInner($value, $analysis_php_version_id, $template_type_map, $type_aliases);
+        $result->offset_start = $offset_start;
+        $result->offset_end = $offset_end;
+        $result->text = $text;
+        $result->from_docblock = $from_docblock;
+        return $result;
+    }
+    /**
+     * @psalm-suppress InaccessibleProperty Allowed during construction
+     *
+     * @param int $analysis_php_version_id contains php version when the type comes from signature
+     * @param array<string, array<string, Union>> $template_type_map
+     * @param array<string, TypeAlias> $type_aliases
+     */
+    private static function createInner(
+        string $value,
+        ?int   $analysis_php_version_id = null,
+        array  $template_type_map = [],
+        array  $type_aliases = [],
+        bool   $from_docblock = false
     ): Atomic {
         switch ($value) {
             case 'int':
```

Comment:
here I suppose every type should be able to receive the $from_docblock, otherwise you removed it before calling create but didn't pass it to the Atomic

Reply:
Yeah I didn't want to cause too much churn and worsen readability, given that I can easily assign `from_docblock` in ::create.